### DISCUSSION TITLE
Janusgraph Cache

### DISCUSSION THREAD

**Discussion by hummingsoul** [simon824/janusgraph-externals#5] [Q&A] [answered]
请教缓存如何设置啊 集成后 服务经常Java heap space内存溢出 分析dump文件 内存占用 前面一堆janus的缓存cache类
不确定是不是缓存的问题 看网上事务缓存的设置 cache.tx-cache-size 也没有一个参考设置
![屏幕截图 2021-01-19 101154](https://user-images.githubusercontent.com/26573994/104979448-40b56000-5a3f-11eb-9eca-e5db45c222f1.jpg)

**Comment by simon824** [comment, marked as answer]
这是 JanusGraph 关于 Cache 的配置和默认值，使用内嵌方式部署的话可以加大一些Java程序的堆内存试试。关于缓存的分类以及更多信息可以参考 [缓存 Caching](https://github.com/simon824/janusgraph.cn/blob/main/JanusGraph%E5%AE%98%E6%96%B9%E6%96%87%E6%A1%A3%E7%BF%BB%E8%AF%91/%E5%9F%BA%E7%A1%80(JanusGraph%20Basics)/%E7%BC%93%E5%AD%98(JanusGraph%20cache).md)

| Name | Description | Datatype | Default Value | Mutability |
| ---- | ---- | ---- | ---- | ---- |
| cache.db-cache | Whether to enable JanusGraph's database-level cache, which is shared across all transactions. Enabling this option speeds up traversals by holding hot graph elements in memory, but also increases the likelihood of reading stale data.  Disabling it forces each transaction to independently fetch graph elements from storage before reading/writing them. | Boolean | false | MASKABLE |
| cache.db-cache-clean-wait | How long, in milliseconds, database-level cache will keep entries after flushing them.  This option is only useful on distributed storage backends that are capable of acknowledging writes without necessarily making them immediately visible. | Integer | 50 | GLOBAL_OFFLINE |
| cache.db-cache-size | Size of JanusGraph's database level cache.  Values between 0 and 1 are interpreted as a percentage of VM heap, while larger values are interpreted as an absolute size in bytes. | Double | 0.3 | MASKABLE |
| cache.db-cache-time | Default expiration time, in milliseconds, for entries in the database-level cache. Entries are evicted when they reach this age even if the cache has room to spare. Set to 0 to disable expiration (cache entries live forever or until memory pressure triggers eviction when set to 0). | Long | 10000 | GLOBAL_OFFLINE |
| cache.tx-cache-size | Maximum size of the transaction-level cache of recently-used vertices. | Integer | 20000 | MASKABLE |
| cache.tx-dirty-size | Initial size of the transaction-level cache of uncommitted dirty vertices. This is a performance hint for write-heavy, performance-sensitive transactional workloads. If set, it should roughly match the median vertices modified per transaction. | Integer | (no default value) | MASKABLE |
